### DISCUSSION TITLE
"isInControl" trigger

### DISCUSSION THREAD

**Discussion by Amidweiz** [ikemen-engine/Ikemen-GO#864] [Feature Request]
Similar to the `IsHost` trigger where it only shows up on one end of a netplay game, the IsInControl trigger would return true if said character is in control of the client and would subsequently return false on the other clients end. The use case for this would be for attacks that are intended to mix up an opponent but show up normally on the players end (E.G. invisibility attack, false clone attack, etc.)

It would also be good if the `GameMode` trigger also could return for replay mode so it you could code said attacks to display differently in said mode. If this hasn't already been implemented into the engine.

**Comment by K4thos** [comment]
2. Strong objection to this, replay is meant to represent exactly what happened during match. GameMode returning something different during replay means broken replays for characters that use GameMode trigger as intended (code meant to fire in particular mode only, affecting the battle, would not start at all during replay).

**Comment by Windblade-GR01** [comment]
Same with K4thos. The `GameMode` change would break replays badly.
So we have to deny that request.

Maybe is just me but I do not understand what advantage would `IsInControl` bring compared to `IsHost`.
It would be posible to give more details about it?

**Reply by Amidweiz** [reply]
Figured I'd bump this thread due to my latest QOL trigger discussion but to answer @Windblade-GR01 `IsInControl` is meant to be a `whoami` like trigger able to produce asynchronous events which the `IsHost` trigger is able to produce without needing to be the host of a match to use them. Also the trigger `IsHomeTeam` along with `pXname` triggers are not able to produce said asynchronous events from testing which I was recommended to try.

**Comment by potsmugen** [comment]
The use cases themselves aren't such bad ideas, but when you really think about it I think fighting games should be designed with local play in mind. You'd go through all that trouble to make invisibility work in netplay, only for it to still not work in local.

**Reply by Amidweiz** [reply]
Obviously said use cases are more intended for online versus play hence why I originally also requested the `gamemode` trigger along side it which did get implemented. So one could determine if they're in an online match compared to local and set up said effects appropriately.
